app/controllers/application_controller.rb

```diff
@@ -308,7 +308,7 @@ def guardian
   end
 
   def current_homepage
-    current_user ? SiteSetting.homepage : SiteSetting.anonymous_homepage
+    current_user&.user_option&.homepage || SiteSetting.anonymous_homepage
```

Comment:
wait, what if homepage is set to nil in user option, it should be homepage then not anonymous_hompage

test with "unread" as the first in the list

Reply:
`homepage` is not the setting. `user_home` is the setting.

`homepage` is a method with [this definition](https://github.com/discourse/discourse/pull/5268/files#diff-162f0f01d64443df28e0ffe2a57be378R131).

Reply:
I see, makes sense